server/service/client.go

```diff
@@ -129,13 +131,24 @@ func (c *Client) doWithHeaders(verb, path, rawQuery string, params interface{},
 	return c.http.Do(request)
 }
 
+// BaseURL returns the parsed base URL of the client as a value, so that the
+// internal URL cannot be modified.
+func (c *Client) BaseURL() url.URL {
+	return *c.baseURL
+}
+
 func (c *Client) Do(verb, path, rawQuery string, params interface{}) (*http.Response, error) {
+	return c.DoContext(context.Background(), verb, path,
+		rawQuery, params)
```

Comment:
Nitpick: should go in 1 line only.

It would be nice to have a very opinionated `gofmt` for silly things like this to not happen.